### PR TITLE
Initial signature aggregator unit tests

### LINKED ISSUE #446: Add unit tests for signature aggregator
**Context and scope**
Currently we are relying only on E2E test for testing functionality throughout the repo. With separating out signature-aggregator and when doing #388 we should add unit tests to ensure that validity checks, aggregation logic etc are valid.

### PR DESCRIPTION
**Pull request by feuGeneA**
## Why this should be merged
it's a first step towards fixing #446

### CODE REVIEW

File: signature-aggregator/aggregator/aggregator.go [hunk 1]
```diff
@@ -343,6 +343,8 @@ func (s *SignatureAggregator) CreateSignedMessage(
 	return nil, errNotEnoughSignatures
 }
 
+// TODO: consider making this function private. its only reference seems to be
```
Comment (iansuvak) [inline signature-aggregator/aggregator/aggregator.go:346]: Yup both of the below should be private. That was my oversight in the move

Comment (feuGeneA) [reply signature-aggregator/aggregator/aggregator.go]: addressed in https://github.com/ava-labs/awm-relayer/pull/457/commits/e5d44b441029b356278d6d13549ff6c4fdb89291

File: signature-aggregator/aggregator/aggregator_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/ava-labs/awm-relayer/signature-aggregator/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+var sigAggMetrics *metrics.SignatureAggregatorMetrics
+var messageCreator message.Creator
+
+func instantiateAggregator(t *testing.T) (
```
Comment (geoff-vball) [inline signature-aggregator/aggregator/aggregator_test.go:22]: Should we create a happy-path test here? Otherwise this PR LGTM

File: .github/workflows/mock_checker.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Check generated code is up to date
```
Comment (minghinmatthewlam) [inline .github/workflows/mock_checker.yml:1]:
```suggestion
name: Mock Checker
```

File: peers/app_request_network.go [hunk 4]
```diff
@@ -1,6 +1,8 @@
 // Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
+//go:generate mockgen -source=$GOFILE -destination=./mocks/mock_app_request_network.go -package=mocks
```
Comment (minghinmatthewlam) [inline peers/app_request_network.go:4]: why is this included in the source file? I would expect the source file to not need to have knowledge of the mock file at all. Instead, can the mock file keep track of command needed to regenerate?

Comment (feuGeneA) [reply peers/app_request_network.go:4]: it's here because the mock is generated from this source file, which is represented on this line by `$GOFILE`. btw i took this from your example 😅 at https://github.com/ava-labs/awm-relayer/pull/21/files#diff-b7c4d6477bfb555408745965840c387dcfc91a74bcbdc3b30d31d594e9abc0e5R4

my understanding is that the mock source file is 100% generated so we can't really put anything in there.

File: signature-aggregator/aggregator/aggregator_test.go (hunk 2 above)
Comment (feuGeneA) [reply signature-aggregator/aggregator/aggregator_test.go:22]: I think yes, but I'd prefer to defer it to a subsequent PR. I'm actively working on it but would like to see this PR get merged since it's already quite close.

**Review by geoff-vball — APPROVED** [review]

**Review by iansuvak — COMMENTED** [review]
Generally LGTM but I do prefer `Mock Checker` that @minghinmatthewlam suggested as the name visible on the UI rather than the more verbose one, it also follows the pattern of `ABI Go Bindings Checker`.

File: scripts/test.sh [hunk 5]
```diff
@@ -27,12 +27,14 @@ if [ "$HELP" = true ]; then
...
+root=$(
     cd "$(dirname "${BASH_SOURCE[0]}")"
     cd .. && pwd
 )
-source "$RELAYER_PATH"/scripts/constants.sh
+source "$root"/scripts/constants.sh
 
 go build -o tests/cmd/decider/decider ./tests/cmd/decider/
 
+"$root"/scripts/generate.sh
```
Comment (iansuvak) [inline scripts/test.sh:38]: I would still rather call this manually than have it be re-generated automatically ever. If it's quick it might not make too much of a difference. not a blocker though

**Review by iansuvak — APPROVED** [review]

File: scripts/test.sh (hunk 5 above)
Comment (cam-schultz) [reply scripts/test.sh:38]: I agree. I wouldn't ever expect running a test to alter my git status
